lib/utils.js

```diff
@@ -475,25 +476,15 @@ function applyWriteConcern(target, sources, options) {
   }
 
   if (options.w != null || options.j != null || options.fsync != null) {
-    const writeConcern = {};
-    if (options.w != null) writeConcern.w = options.w;
-    if (options.wtimeout != null) writeConcern.wtimeout = options.wtimeout;
-    if (options.j != null) writeConcern.j = options.j;
-    if (options.fsync != null) writeConcern.fsync = options.fsync;
+    const writeConcern = new WriteConcern(options.w, options.wtimeout, options.j, options.fsync);
     return Object.assign(target, { writeConcern });
   }
 
-  if (
-    coll &&
-    (coll.writeConcern.w != null || coll.writeConcern.j != null || coll.writeConcern.fsync != null)
-  ) {
+  if (coll && coll.writeConcern && coll.writeConcern.isDefined()) {
```

Comment:
even with the current approach you don't need to check `isDefined` here, because the getter already checks and returns `null` otherwise.